core/state_transition.go

```diff
@@ -615,7 +615,8 @@ func (st *stateTransition) innerExecute() (*ExecutionResult, error) {
 			if len(st.msg.Data) > 0 {
 				var logs []*types.Log
 				// run the storage transaction
-				logs, vmerr = storagetx.ExecuteTransaction(st.msg.Data, st.msg.BlockNumber, st.msg.TransactionHash, msg.From, st.evm.StateDB)
+				// We set the tx index to 0, since it doesn't matter because this execution won't modify the account state
+				logs, vmerr = storagetx.ExecuteTransaction(st.msg.Data, st.msg.BlockNumber, st.msg.TransactionHash, 0, msg.From, st.evm.StateDB)
```

Comment:
@draganm AFAICT, this is the call that is used for things like `estimateGas`, so it doesn't actually modify the account state. So I think we can just pass a default value here for the transaction index and we don't need to bother trying to get the real index (we can get the real index if needed, but it requires changes at all call sites).